tests/unit/backend/models/form.server.model.spec.ts

```diff
@@ -872,13 +870,105 @@ describe('Form Model', () => {
         const expectedOtpData = {
           form: form._id,
           formAdmin: {
-            email: preloadedAdmin.email,
-            userId: preloadedAdmin._id,
+            email: populatedAdmin.email,
+            userId: populatedAdmin._id,
           },
           msgSrvcName: encryptFormParams.msgSrvcName,
         }
         expect(actualOtpData).toEqual(expectedOtpData)
       })
     })
   })
+
+  describe('getDashboardForms', () => {
+    it('should return empty array when user has no forms to view', async () => {
+      // Arrange
+      const randomUserId = new ObjectId()
+      const invalidEmail = 'not-valid@example.com'
+
+      // Act
+      const actual = await Form.getDashboardForms(randomUserId, invalidEmail)
+
+      // Assert
+      expect(actual).toEqual([])
+    })
+
+    it('should return array of forms user is permitted to view', async () => {
+      // Arrange
+      // Add additional user.
+      const differentUserId = new ObjectId()
+      const diffPreload = await dbHandler.insertFormCollectionReqs({
+        userId: differentUserId,
+        mailName: 'something-else',
+        mailDomain: MOCK_ADMIN_DOMAIN,
+      })
+      const diffPopulatedAdmin = merge(diffPreload.user, {
+        agency: diffPreload.agency,
+      })
+      // Populate multiple forms with different permissions.
+      // Is admin.
+      const userOwnedForm = await Form.create(MOCK_EMAIL_FORM_PARAMS)
+      // Has permissions.
+      const userPermissionForm = await Form.create({
+        ...MOCK_ENCRYPTED_FORM_PARAMS,
+        admin: diffPopulatedAdmin._id,
+        permissionList: [{ email: populatedAdmin.email, write: true }],
+      })
```

Comment:
lol, I wonder if it might be better to separate these tests, eg one test each for admin, write permissions, read permissions, archived, no permissions

Reply:
eh, this is a base function that is supposed to return all the above types. so keeping it all in one test works too since it will fail if any of the types does not get returned. 

building on your comment, will make the test more inclusive, add write permissions test form too just in case that fails

Reply:
I know it works, I'm just trying to improve the readability and modularity of the tests